src/edu/washington/escience/myria/daemon/MyriaDriverLauncher.java

```diff
@@ -229,55 +228,60 @@ private static void logStartupMessage(Class<? extends Name<?>>[] classes, Inject
       final String doc = annotation.doc();
       final String defaultVal;
       {
-        if (!annotation.default_value().equals(NamedParameter.REEF_UNINITIALIZED_VALUE))
+        if (!annotation.default_value().equals(NamedParameter.REEF_UNINITIALIZED_VALUE)) {
           defaultVal = annotation.default_value();
-        else if (!annotation.default_class().equals(Void.class))
+        } else if (!annotation.default_class().equals(Void.class)) {
           defaultVal = annotation.default_class().getSimpleName();
-        else if (annotation.default_values().length > 0)
+        } else if (annotation.default_values().length > 0) {
           defaultVal = Arrays.toString(annotation.default_values());
-        else if (annotation.default_classes().length > 0) {
-          Class<?>[] default_classes = annotation.default_classes();
-          String[] sss = new String[default_classes.length];
-          for (int i1 = 0; i1 < default_classes.length; i1++) {
-            sss[i1] = default_classes[i1].getSimpleName();
+        } else if (annotation.default_classes().length > 0) {
+          StringBuilder b = new StringBuilder("[");
+          Class<?>[] cls = annotation.default_classes();
+          for (int j = 0; ; j++) {
+
+            b.append(cls[j].getSimpleName());
+            if (j == cls.length - 1) {
+              b.append(']');
+              break;
+            }
+            b.append(", ");
           }
-          defaultVal = Arrays.toString(sss);
+          defaultVal = b.toString();
         } else defaultVal = "";
       }
 
-      if (i > 0) sb.append('\n');
+      if (i > 0) {
```

Comment:
This strongly suggests that you can just append each line to a list and generate a newline-separated message with `Joiner.on("\n")`. That would help expose intention (high-level structure of the message) rather than mechanism (appending each delimiter one-at-a-time to a `StringBuilder`).